lib/identity_doc_auth/response.rb

```diff
@@ -11,13 +11,16 @@ def initialize(success:, errors: {}, exception: nil, extra: {}, pii_from_doc: {}
     end
 
     def merge(other)
-      Response.new(
-        success: success? && other.success?,
-        errors: errors.merge(other.errors),
-        exception: exception || other.exception,
-        extra: extra.merge(other.extra),
-        pii_from_doc: pii_from_doc.merge(other.pii_from_doc),
-      )
+      # To maintain the subclass and avoid mutating the instance, set ivars on a clone.
+      clone = self.clone
+      clone.instance_eval do
```

Comment:
what if we did `self.class.new` instead to preserve the class?

Reply:
The problem is the constructor argument signature is entirely different for almost all of the Acuant subclasses, in a way that's pretty hard to reconcile without rewriting most of them 🤷 

Compare:

https://github.com/18F/identity-doc-auth/blob/60ac42fd52976dd756cf58246af51009e9cf0aa6/lib/identity_doc_auth/response.rb#L5

https://github.com/18F/identity-doc-auth/blob/60ac42fd52976dd756cf58246af51009e9cf0aa6/lib/identity_doc_auth/acuant/responses/get_results_response.rb#L14

Reply:
oof, these definitely violate SOLID principals...

I feel like we wrote `#merge` so that the IDP's `FormResponse` could merge with these.... I wonder how badly these breaks those assumptions

Another thought:
- maybe we make sure that every subclass that overrides the initializer should override `#merge` too?

Reply:
confirmed, we did assume that `FormResponse` and `DocAuth::Response` could merge: https://github.com/18F/identity-idp/blob/315ccab0f34a089775fc3947f2f90035a928ff81/spec/services/form_response_spec.rb#L43-L69

Reply:
> * maybe we make sure that every subclass that overrides the initializer should override `#merge` too?

Yeah, I wondered about this as well. But even within the class itself, it's hard to know what to pass to its own constructor, since something like `http_response` is a complex argument, and there's not a clear answer to what a "merged" HTTP response should contain.

I'll think on it a bit longer. @solipet and I had chatted about these as well, and while we weren't entirely happy with this solution, it seemed the most reasonable compromise.

In an earlier iteration I'd also skipped the need for `instance_eval` and just mutated the original instance, renaming the method to `merge!`. I was a bit worried about the downstream considerations of this kind of change, but open to reconsidering it if it's a more agreeable option.

Reply:
> confirmed, we did assume that `FormResponse` and `DocAuth::Response` could merge: 

This just gets more and more thorny 😅  

Yet another option I considered is: Maybe we don't maintain the subclass and keep it as it is returning the base class. Then, we'd need to avoid any customizations to those subclasses beyond what's assigned in the constructor, so that the properties / methods available between the base class and the subclasses are consistent.

Reply:
Ok so it seems to me like the core issue here is the `#to_h` override, not the rest of this, right? So as long as that gets fixed, the rest of the PR is probably fine

Reply:
As discussed on Slack, since the removal of the Acuant-specific `to_h` override is the main fix here, I might plan to back out the changes to `merge`.

I'll probably keep most of the spec addition, since it would have been nice to have existed anyways.

Reply:
> As discussed on Slack, since the removal of the Acuant-specific `to_h` override is the main fix here, I might plan to back out the changes to `merge`.

Updated in b50a338.